tests/docs/multi_dut_asic_testing_support.md

```diff
@@ -0,0 +1,499 @@
+duthosts["node1"].foo(asic_index="all")
+
+duthosts.nodes[1].foo()                 # Equivalent to: duthosts[1].foo()
+duthosts.nodes[1].foo(asic_index=1)     # Equivalent to: duthosts[1].foo(asic_index=1)
+duthosts.nodes[1].asics[1].foo()        # Equivalent to: duthosts[1].asics[1].foo()
+duthosts.nodes[1].foo(asic_index="all") # Equivalent to: duthosts[1].foo(asic_index="all")
+
+duthosts.nodes["node1"].foo()                 # Equivalent to: duthosts["node1"].foo()
+duthosts.nodes["node1"].foo(asic_index=1)     # Equivalent to: duthosts["node1"].foo(asic_index=1)
+duthosts.nodes["node1"].asics[1].foo()        # Equivalent to: duthosts["node1"].asics[1].foo()
+duthosts.nodes["node1"].foo(asic_index="all") # Equivalent to: duthosts["node1"].foo(asic_index="all")
+
+duthosts.frontend_nodes.foo()
+duthosts.frontend_nodes.foo(asic_index=1)
+duthosts.frontend_nodes.foo(asic_index="all")
+
+[node.foo() for node in duthosts.frontend_nodes]
+
+duthosts.frontend_nodes[1].foo()
+duthosts.frontend_nodes[1].foo(asic_index=1)
+duthosts.frontend_nodes[1].asics[1].foo()  # Equivalent to the line above
+duthosts.frontend_nodes[1].foo(asic_index="all")
+
+duthosts.frontend_nodes["node1"].foo()
+duthosts.frontend_nodes["node1"].foo(asic_index=1)
+duthosts.frontend_nodes["node1"].asics[1].foo()  # Equivalent to the line above
+duthosts.frontend_nodes["node1"].foo(asic_index="all")
+
+duthosts.supervisor_nodes.bar()
+duthosts.supervisor_nodes[0].bar()
+duthosts.supervisor_nodes["supervisor0"].bar()
+```
+
+In the above examples:
+* `duthost.asics` should be a list of objects. Each object represents an ASIC in the DUT. `duthost.asics` represents all the ASIC instances in the DUT.
+* `duthosts.nodes` should be a list of objects. Each object represents a DUT host in the multi-DUT system.
+* `duthosts.frontend_nodes` should be a subset of `duthosts.nodes`. Each object represent a frontend DUT host in multi-DUT system.
+* `duthosts.supervisor_nodes` should be a subset of `duthosts.nodes`. Each node represents a supervisor DUT host in multi-DUT system.
+* The `duthosts` should can be extended with other attributes to represent other specific type of nodes. For example `duthosts.supervisor_nodes`.
+
+
+### Backward compatibility
+```python
+duthost.foo()
+```
+The `duthost` fixture should support running operations on single DUT just like before. Call like `duthost.foo()` should be applicable for both single-ASIC and multi-ASIC DUT. For single-ASIC DUT, everything should be the same as before.
+
+### Multi-ASIC related operations
+```python
+duthost.foo(asic_index=1)
+duthost.asics[1].foo()      # Equivalent to the line above
+duthost.foo(asic_inex="all")
+[node.foo() for node in duthosts.nodes]
+```
+For multi-ASIC related operations on DUT host, the function or ansible module call should accept optional `asic_index` argument.
+* No `asic_index`, perform operation just on the host (or default namespace). For example, just run command `"show ip bgp summary"`.
```

Comment:
This is a bit misleading. It really depends on the CLIs. In this example "show ip bgp summary" CLI walks through all the ASIC namespaces and does not return the 'bgp summary' from default namespace.

Reply:
Sure, I'll update it to make it more clear.